tests/src/test/scala/whisk/core/container/test/ContainerUtilsTests.scala

```diff
@@ -0,0 +1,147 @@
+@RunWith(classOf[JUnitRunner])
+class ContainerUtilsTests extends FlatSpec
+    with BeforeAndAfter
+    with Matchers
+    with StreamLogging
+    with TimeLimitedTests {
+
+    val timeLimit = 100 millis
+
+    before {
+        stream.reset()
+    }
+
+    implicit val transid = TransactionId.testing
+
+    behavior of "getDockerLogContent"
+
+    case class FixtureParam(file: File, writer: FileWriter, cu: ContainerUtilsTester)
+
+    def withFixture(test: OneArgTest) = {
+        val file = File.createTempFile(this.getClass.getName, test.name.replaceAll("[^a-zA-Z0-9.-]", "_"))
+        val writer = new FileWriter(file)
+        val cu = new ContainerUtilsTester(file)
+
+        val fixture = FixtureParam(file, writer, cu)
+
+        try {
+            withFixture(test.toNoArgTest(fixture))
+        } finally {
+            writer.close()
+            file.delete()
+        }
+    }
+
+    def writeLogFile(fixture: FixtureParam, content: String): Long = {
+        fixture.writer.write(content)
+        fixture.writer.flush()
+        fixture.file.length
+    }
+
+    it should "tolerate an empty log file" in { fixture =>
+        val logText = ""
+        val size = writeLogFile(fixture, logText)
+
+        val buffer = fixture.cu.getDockerLogContent(containerHash = ContainerHash.fromString("0123"), start = 0, end = size, mounted = false)
+        val logContent = new String(buffer.array, buffer.arrayOffset, buffer.position, StandardCharsets.UTF_8)
+
+        logContent shouldBe logText
+        stream.size() shouldBe 0
+    }
+
+    it should "read a full log file" in { fixture =>
+        val logText = "text"
+        val size = writeLogFile(fixture, logText)
+
+        val buffer = fixture.cu.getDockerLogContent(containerHash = ContainerHash.fromString("0123"), start = 0, end = size, mounted = false)
+        val logContent = new String(buffer.array, buffer.arrayOffset, buffer.position, StandardCharsets.UTF_8)
+
+        logContent shouldBe logText
+        stream.size() shouldBe 0
+    }
+
+    it should "read a log file portion" in { fixture =>
+        val count = 4
+        val logText = (1 to count).map(i => s"Line ${i}\n").mkString("")
+        val size = writeLogFile(fixture, logText)
+        val quarter = count / 4
+        val qlen = logText.length / 4 // length of a quarter
+        val from = qlen // start reading from second quarter
+        val to = from + 2 * qlen // read second and third quarter
+        val expectedText = logText.substring(from, to)
+
+        val buffer = fixture.cu.getDockerLogContent(containerHash = ContainerHash.fromString("0123"), start = from, end = to, mounted = false)
+        val logContent = new String(buffer.array, buffer.arrayOffset, buffer.position, StandardCharsets.UTF_8)
+
+        logContent shouldBe expectedText
+        stream.size() shouldBe 0
+    }
+
+    it should "tolerate premature end of log file" in { fixture =>
+        val logText = (1 to 2).map(i => s"Line ${i}\n").mkString("")
```

Comment:
Your proposal does only add `\n` between the strings included in the range. In other words: with your proposal, the last line won't be followed by a `\n`.